Add affected attribute format (#175)
pypa/advisory-database: README.md

## Using this data

### Marking specific attributes
It can be helpful to know which specific code elements of a package are vulnerable and this is done by appending an attribute and list of module paths to that attribute to the OSV payload. Eg. 

> ```

Reviewer: Nitpick: syntax highlighting

```suggestion
```json
```

(and similar below)

> It can be helpful to know which specific code elements of a package are vulnerable and this is done by appending an attribute and list of module paths starting from the top level module of a package to the OSV payload. Eg.  

Reviewer 2: Can we mention explicitly this can be set in [ecosystem_specific](https://ossf.github.io/osv-schema/#affectedecosystem_specific-field) (and that this section is defining the format for that)? 

e.g.

> "OSV entries in this database have the following [`ecosystem_specific`](https://ossf.github.io/osv-schema/#affectedecosystem_specific-field) definition: 

```
"ecosystem_specific": {
  "imports": [
    { 
       "attribute": string,
       "modules": [ string ],
    }
  ]
}
```

> "imports" is a JSON array containing the modules and attributes affected by the vulnerability...
> For example, a vulnerability that affects `PIL::ImageFont` can be represented as...

This follows the "informal schema definition" convention of the OSV schema. I've also used "imports" for the top level field name, but there might be better names here.

Author: That seems reasonable enough to me. Pushed an update with your changes 👍

{

>   attribute: "ImageFont",
>   modules: ["PIL"]

Reviewer: Nitpick: JSON syntax since this'll be part of an OSV payload

```suggestion
  "attribute": "ImageFont",
  "modules": ["PIL"]
```

(and similar below)

}
```
which is equivalent to `PIL:ImageFont`. If a second attribute `ImageFont2` is also affected then a second payload should be added and delimited with a `;`. Eg.
```
{
  attribute: "ImageFont",
  modules: ["PIL"];
  attribute: "ImageFont2",
  modules: ["PIL"]
}

> ```

Reviewer: 0.02c: I don't think we should invent custom syntax at the serialization level here. If our goal is to express the dot product, maybe both can just be arrays?

```json
{
  "attributes": ["ImageFont", "ImageFont2"],
  "modules": ["PIL"]
}
```

Reviewer: (This might have negative consequences I haven't thought of yet.)

Another option would be to have this represented as an array of these objects, e.g.:

```json
[
  { "attribute": "ImageFont", "modules": ["PIL"] },
  { "attribute": "ImageFont2", "modules": ["PIL"] }
]
```

Author: I should have probably used a better example here, but the idea is that the two should be completely independent (code, paths, etc...) aside from the top level module. I prefer option two of your suggestions for this, but @oliverchang what do you think best fits the osv style?

Reviewer 2: +1 to not inventing custom syntax. We should keep this as plain JSON.

```
[
  { "attribute": "ImageFont", "modules": ["PIL"] },
  { "attribute": "ImageFont2", "modules": ["PIL"] }
]
```

as @woodruffw suggested seems pretty good to me!

## Using this data

### Marking specific attributes
It can be helpful to know which specific code elements of a package are vulnerable and this is done by appending an attribute and list of module paths starting from the top level module of a package to the OSV payload. Eg. 
OSV entries in this database have the following ecosystem_specific definition: 
```json
"ecosystem_specific": {
  "imports": [
    { 
       "attribute": string,
       "modules": [ string ],
    }
  ]
}
```
"imports" is a JSON array containing the modules and attributes affected by the vulnerability...
For example, a vulnerability that affects PIL::ImageFont can be represented as...
```json
"imports": [
  {
    "attribute": "ImageFont",
    "modules": ["PIL"]
  }
]
```

> which is equivalent to `PIL:ImageFont`. If a second attribute `ImageFont2` is also affected then a second payload should be added and delimited with a `;`. Eg.

Reviewer 2: ```suggestion
which is equivalent to `PIL:ImageFont`. If a second attribute `ImageFont2` is also affected, then a second import entry needs to be added to the `imports` array. 
```

```json
"imports": [
  { "attribute": "ImageFont", "modules": ["PIL"] },
  { "attribute": "ImageFont2", "modules": ["PIL"] }
]
```

> attributes which are accessible via multiple paths may be represented in a condensed form. Consider the attribute `django.db.models:JSONField` from the [django project](https://github.com/django/django/blob/0ee2b8c326d47387bacb713a3ab369fa9a7a22ee/django/db/models/__init__.py#L99) 

Reviewer 2: ```suggestion
Attributes which are accessible via multiple paths may be represented in a condensed form. Consider the attribute `django.db.models:JSONField` from the [django project](https://github.com/django/django/blob/0ee2b8c326d47387bacb713a3ab369fa9a7a22ee/django/db/models/__init__.py#L99) 
```
